src/rdf4cpp/IRIFactory.cpp

```diff
@@ -226,7 +229,35 @@ nonstd::expected<IRI, IRIFactoryError> IRIFactory::create_and_validate(std::stri
     return IRI::make_unchecked(iri, node_storage);
 }
 
-void IRIFactory::assign_prefix(std::string_view prefix, std::string_view expanded) {
+IRIFactoryError IRIFactory::assign_prefix(std::string_view prefix, std::string_view expanded) {
+    using namespace util::char_matcher_detail;
+    static constexpr auto first_matcher = ASCIINumMatcher{} | PNCharsUMatcher;
```

Comment:
Shouldn't this be `PNCharsBaseMatcher`? In the grammar the first char is of `PN_CHARS_BASE`